docs/pages/aggregate.md

```diff
@@ -559,6 +557,71 @@ final class Hotel extends AggregateRoot
 }
 ```
 
+## Working with dates
+
+An aggregate should always be deterministic. In other words, whenever I execute methods on the aggregate, 
+I always get the same result. This also makes testing much easier.
+
+But that often doesn't seem to be possible, e.g. if you want to save a createAt date. 
+But you can pass this information by yourself.
+
+```php
+use Patchlevel\EventSourcing\Aggregate\AggregateRoot;
+use Patchlevel\EventSourcing\Attribute\Aggregate;
+use Patchlevel\EventSourcing\Attribute\Apply;
+
+#[Aggregate('profile')]
+final class Profile extends AggregateRoot
+{
+    private string $id;
+    private Name $name;
+    private DateTimeImmutable $registeredAt;
+    
+    public static function register(string $id, string $name, DateTimeImmutable $registeredAt): static
+    {
+        $self = new static();
+        $self->recordThat(new ProfileRegistered($id, $name, $registeredAt));
+
+        return $self;
+    }
+    
+    // ...
+}
+```
+
+But if you still want to make sure that the time is now and not in the past or future, you can pass a clock.
+
+```php
+use Patchlevel\EventSourcing\Aggregate\AggregateRoot;
+use Patchlevel\EventSourcing\Attribute\Aggregate;
+use Patchlevel\EventSourcing\Attribute\Apply;
+use Patchlevel\EventSourcing\Clock\Clock;
+
+#[Aggregate('profile')]
+final class Profile extends AggregateRoot
+{
+    private string $id;
+    private Name $name;
+    private DateTimeImmutable $registeredAt;
+    
+    public static function register(string $id, string $name, Clock $clock): static
+    {
+        $self = new static();
+        $self->recordThat(new ProfileRegistered($id, $name, $clock->now()));
+
+        return $self;
+    }
+    
+    // ...
+}
+```
+
+This gives you the option of implementing the clock, e.g. use the FrozenClock to always get the same time.
```

Comment:
Maybe add a note for testing here in this context. To prevent ppl. from thinking this should be done in production with the FrozenClock :D